patches/server/0918-use-BlockFormEvent-for-mud-converting-into-clay.patch

```diff
@@ -0,0 +1,28 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Legitimoose <legitimoose@gmail.com>
+Date: Tue, 27 Sep 2022 20:24:58 -0700
+Subject: [PATCH] use BlockFormEvent for mud converting into clay
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java b/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
+index a9a2fb530f6a451fc24d3e164364540473ad2c09..a3540107eb1d9393e7ec8125c2b246a166009f4a 100644
+--- a/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/PointedDripstoneBlock.java
+@@ -213,10 +213,13 @@ public class PointedDripstoneBlock extends Block implements Fallable, SimpleWate
+                             if (((PointedDripstoneBlock.FluidInfo) optional.get()).sourceState.is(Blocks.MUD) && fluidtype == Fluids.WATER) {
+                                 BlockState iblockdata1 = Blocks.CLAY.defaultBlockState();
+ 
+-                                world.setBlockAndUpdate(((PointedDripstoneBlock.FluidInfo) optional.get()).pos, iblockdata1);
+-                                Block.pushEntitiesUp(((PointedDripstoneBlock.FluidInfo) optional.get()).sourceState, iblockdata1, world, ((PointedDripstoneBlock.FluidInfo) optional.get()).pos);
+-                                world.gameEvent(GameEvent.BLOCK_CHANGE, ((PointedDripstoneBlock.FluidInfo) optional.get()).pos, GameEvent.Context.of(iblockdata1));
+-                                world.levelEvent(1504, blockposition1, 0);
++                                // Paper start
++                                if (org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockFormEvent(world, ((PointedDripstoneBlock.FluidInfo) optional.get()).pos, iblockdata1)) {
++                                    Block.pushEntitiesUp(((PointedDripstoneBlock.FluidInfo) optional.get()).sourceState, iblockdata1, world, ((PointedDripstoneBlock.FluidInfo) optional.get()).pos);
```

Comment:
Generally, refain from changing the indentation of vanilla code where tenable, just helps reduce diff/merge noise

Reply:
Noted, thanks

Reply:
(if you could revert the indentation changes and modify the patch that would be great!)

Reply:
Amended the commit. Indentation should be fixed now if I did this correctly!